### PR TITLE
feat: add card form

### PR DESCRIPTION
**Pull request by ahmaxed**
Checklist:

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/components/Donation/DonateForm.tsx [hunk 1]
```diff
@@ -99,7 +100,7 @@ const mapDispatchToProps = {
   postChargeStripe
 };
 
-class DonateForm extends Component<DonateFormProps, DonateFormState> {
+class DonateForm extends Component<DonateFormProps, DonateFromComponentState> {
```
Comment (ShaunSHamilton) [inline client/src/components/Donation/DonateForm.tsx:103]:
```suggestion
class DonateForm extends Component<DonateFormProps, DonateFormComponentState> {
```

File: client/src/components/Donation/DonateForm.tsx [hunk 2]
```diff
@@ -48,12 +42,19 @@ const numToCommas = (num: number): string =>
...
   redirecting: boolean;
   success: boolean;
   error: string;
+  loading: {
+    stripe: boolean;
+    paypal: boolean;
+  };
+};
+
+type DonateFromComponentState = {
```
Comment (ShaunSHamilton) [inline client/src/components/Donation/DonateForm.tsx:55]:
```suggestion
type DonateFormComponentState = {
```

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Just because it caught my eye

**Comment by ahmadabdolsaheb** [comment]
We are moving in another direction and as a result we would not be integrating square for the time being.

**Pull request closed without merging**